docs: document CI versions of Ubuntu (#13625)
Homebrew/brew: docs/Linux-CI.md

# Linux CI in `homebrew/core`

We currently use Ubuntu 22.04 for bottling in `homebrew/core`.

### Ubuntu versus other distributions

As of 2022, around 77% of our users are using Ubuntu. This is the reason why we have chosen ubuntu for our base CI image.
We have been using Ubuntu for CI since version 14.04 with success.
The LTS versions have 5 year support. A new LTS version is released every 2 years.

> In theory this gives us 5 year to migrate to the next version, but we aim to migrate to the newest version within the 2 years before yet another version is published.

Reviewer: This is a little confusing. Can we reword this?

> Our bottles are compatible with other distributions like debian/centos, even if compiled with Ubuntu.

Reviewer 2: ```suggestion
Our bottles are compatible with other distributions like Debian/CentOS, even if compiled with Ubuntu.
```

### Past and future migrations

> 

Reviewer 2: ```suggestion
```

> | Distribution | Glibc | GCC | Usage |

Reviewer 2: The table is not rendering. At least for me in GH UI.

# Linux CI in `homebrew/core`

We currently use Ubuntu 16.04 for bottling in `homebrew/core`.

### Ubuntu versus other distributions

As of 2022, around 77% of our users are using Ubuntu. This is the reason why we have chosen this distribution for our base CI image.
We have been using Ubuntu for CI since version 14.04 with success.
The LTS versions have 5 year support. A new LTS version is released every 2 years.

Our bottles are compatible with other distributions like Debian/CentOS, even when compiled on Ubuntu.

### Next versions

We are currently moving our CI to Ubuntu 20.04. This work will probably be done before end of 2022.

> Moving from Ubuntu 16.04 to Ubuntu 20.04 (and thus skipping 18.04) took more time than expected due to other more urgent issues we had to take care of.

Reviewer: I thought we were doing 22.04 given glibc 2.35?

Reviewer 3: Oh, yes, I misread this and have this question too.

Reviewer 4: The majority of our users are using Ubuntu 20.04 and Glibc 2.31. I suggest that we build bottles using Ubuntu 20.04 and Glibc 2.31. 79% of our users (23% on Glibc 2.35 and 56% on Glibc 2.31) will be able to use their host's Glibc and compiler without needing to installed the brewed Glibc and GCC.

We plan to proceed with a more regular update from 2022 on. We aim to use the latest - 1 Ubuntu LTS version for our CI.

> The 2 years delay gives our users time to upgrade, and reduces the number of users that will have to install Homebrew's `glibc`.

Reviewer: ```suggestion
The two-year delay gives our users time to upgrade, and reduces the number of users that will have to install Homebrew's `glibc`.
```
